client/mds/infra-connector/service.go

```diff
@@ -31,3 +39,152 @@ func (s *Service) GetRegionsWithDataPlanes(regionsQuery *DataPlaneRegionsQuery)
+}
+
+func (s *Service) GetCertificates(query *MDSCertificateQuery) (model.Paged[model.MdsByocCertificate], error) {
+	var response model.Paged[model.MdsByocCertificate]
+	if query == nil {
+		return response, fmt.Errorf("query cannot be nil")
+	}
+
+	reqUrl := fmt.Sprintf("%s/%s", s.Endpoint, Certificate)
+
+	if query.Size == 0 {
+		query.Size = defaultPage.Size
+	}
+
+	_, err := s.Api.Get(&reqUrl, query, &response)
+	if err != nil {
+		return response, err
+	}
+	return response, nil
+}
+
+func (s *Service) GetTshirtSizes(query *MdsTshirtSizesQuery) (model.Paged[model.MdsByocTshirtSize], error) {
+	var response model.Paged[model.MdsByocTshirtSize]
+	if query == nil {
+		return response, fmt.Errorf("query cannot be nil")
+	}
+
+	reqUrl := fmt.Sprintf("%s/%s/%s", s.Endpoint, K8sCluster, TshirtSize)
+
+	if query.Size == 0 {
+		query.Size = defaultPage.Size
+	}
+
+	_, err := s.Api.Get(&reqUrl, query, &response)
+	if err != nil {
+		return response, err
+	}
+	return response, nil
+}
+
+func (s *Service) GetProviderTypes() ([]string, error) {
+	urlPath := fmt.Sprintf("%s/%s/%s", s.Endpoint, CloudAccount, Types)
+	var response []string
+
+	_, err := s.Api.Get(&urlPath, nil, &response)
+	if err != nil {
+		return response, err
+	}
+
+	return response, err
+}
+
+func (s *Service) GetCloudProviderRegions() ([]model.MdsByocCloudProviderRegion, error) {
+	var response []model.MdsByocCloudProviderRegion
+
+	reqUrl := fmt.Sprintf("%s/%s", s.Endpoint, CloudProviders)
+
+	_, err := s.Api.Get(&reqUrl, nil, &response)
+	if err != nil {
+		return response, err
+	}
+	return response, nil
+}
+
+// CreateDataPlane - Submits a request to create dataplane
+func (s *Service) CreateDataPlane(requestBody *ByocDataPlaneCreateRequest) (*model.MdsPolicy, error) {
+	if requestBody == nil {
+		return nil, fmt.Errorf("requestBody cannot be nil")
+	}
+	var response model.MdsPolicy
+	urlPath := fmt.Sprintf("%s/%s", s.Endpoint, K8sCluster)
+
+	_, err := s.Api.Post(&urlPath, requestBody, &response)
+	if err != nil {
+		return &response, err
+	}
+
+	return &response, err
+}
+
+func (s *Service) GetDataPlane(id string) ([]model.MdsByocCloudProviderRegion, error) {
+	var response []model.MdsByocCloudProviderRegion
+
+	reqUrl := fmt.Sprintf("%s/%s/%s", s.Endpoint, K8sCluster, id)
+
+	_, err := s.Api.Get(&reqUrl, nil, &response)
+	if err != nil {
+		return response, err
+	}
+	return response, nil
+}
+
+func (s *Service) GetDataPlaneByName(query *ByocDataPlaneQuery) (model.Paged[model.ByocDataPlane], error) {
+	urlPath := fmt.Sprintf("%s/%s", s.Endpoint, K8sCluster)
+	var response model.Paged[model.ByocDataPlane]
+
+	if query.Size == 0 {
+		query.Size = defaultPage.Size
+	}
+
+	_, err := s.Api.Get(&urlPath, query, &response)
+	if err != nil {
+		return response, err
+	}
+
+	return response, nil
+}
+
+func (s *Service) GetDataPlaneById(id string) (model.ByocDataPlane, error) {
+	urlPath := fmt.Sprintf("%s/%s/%s", s.Endpoint, K8sCluster, id)
+	var response model.ByocDataPlane
```

Comment:
same here, return type should be universal, just `DataPlane`. And method name only `GetDataPlane`